src/state/slices/opportunitiesSlice/resolvers/osmosis/index.ts

```diff
@@ -0,0 +1,114 @@
+import type { ToAssetIdArgs } from '@shapeshiftoss/caip'
+import { osmosisChainId, toAssetId } from '@shapeshiftoss/caip'
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import type { ReduxState } from 'state/reducer'
+import type { PortfolioAccountBalancesById } from 'state/slices/portfolioSlice/portfolioSliceCommon'
+import {
+  selectAssetById,
+  selectPortfolioAccountBalances,
+  selectPortfolioLoadingStatusGranular,
+} from 'state/slices/selectors'
+
+import type {
+  GetOpportunityIdsOutput,
+  GetOpportunityMetadataOutput,
+  LpId,
+  OpportunityMetadata,
+} from '../../types'
+import { toOpportunityId } from '../../utils'
+import type { OpportunitiesMetadataResolverInput, OpportunityUserDataResolverInput } from '../types'
+import { generateAssetIdFromOsmosisDenom, getPools } from './utils'
+
+export const osmosisLpOpportunitiesMetadataResolver = async ({
+  opportunityType,
+  reduxApi,
+}: OpportunitiesMetadataResolverInput): Promise<{ data: GetOpportunityMetadataOutput }> => {
+  const { getState } = reduxApi
+  const state: any = getState()
+
+  const lpOpportunitiesById: Record<LpId, OpportunityMetadata> = {}
+
+  const liquidityPools = await getPools()
```

Comment:
We should pass IDs around. See `OpportunitiesMetadataResolverInput

This way we can add chainId safety at the beginning of this resolver and before this gets it - see https://github.com/shapeshift/web/pull/3614 - which means this will only be called when the resolver is called IBC chain IDs, reducing the network many many x.

See 3614 for context

Reply:
👍